crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3059,7 +3073,90 @@ async fn get_bank_account_connector_details(
         None => Ok(None),
     }
 }
+pub async fn set_default_payment_method(
+    state: routes::AppState,
+    merchant_account: domain::MerchantAccount,
+    key_store: domain::MerchantKeyStore,
+    customer_id: &str,
+    payment_method_id: String,
+) -> errors::RouterResponse<CustomerDefaultPaymentMethodResponse> {
+    let db = &*state.store;
+    //check for the customer
+    let customer = db
+        .find_customer_by_customer_id_merchant_id(
+            customer_id,
+            &merchant_account.merchant_id,
+            &key_store,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::CustomerNotFound)?;
+    // check for the presence of payment_method
+    let payment_method = db
+        .find_payment_method(&payment_method_id)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::PaymentMethodNotFound)?;
+    if payment_method.customer_id != customer_id {
+        Err(errors::ApiErrorResponse::PreconditionFailed {
+            message: "The payment_method_id is not valid for the Customer".to_string(),
+        })
+        .into_report()?
+    }
```

Comment:
We need to check merchant ID also, because currently, customer IDs can be duplicated across merchants. Also, use the `utils::when()` function.